apps/near_app/near_api.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file    near_api.h
+ * @author  Cypherock X1 Team
+ * @brief   Header file to export some helper functions for the NEAR app
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ * target=_blank>https://mitcc.org/</a>
+ */
+#ifndef NEAR_API_H
+#define NEAR_API_H
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include <near/core.pb.h>
+#include <stdint.h>
+
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * EXPORTED VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief API to decode query from host with `NEAR_QUERY_FIELDS`
+ *
+ * @param[in] data: PB encoded bytestream received from host
+ * @param[in] data_size: size of pb encoded bytestream
+ * @param[out] query_out: @ref near_query_t obj to copy the decoded result to
+ * @return bool True if decoding was successful, else false
+ */
+bool decode_near_query(const uint8_t *data,
+                       uint16_t data_size,
+                       near_query_t *query_out);
+
+/**
+ * @brief Encodes the NEAR result with `NEAR_RESULT_FIELDS` to byte-stream
+ *
+ * @param[in] result: object of populated @ref near_result_t to be encoded
+ * @param[out] buffer: buffer to fill byte-stream into
+ * @param[in] max_buffer_len: Max length allowed for writing bytestream to
+ * buffer
+ * @param[out] bytes_written_out: bytes written to bytestream
+ * @return bool True if decoding was successful, else false
+ */
+bool encode_near_result(near_result_t *result,
```

Comment:
let's make the parameter `const`

```suggestion
bool encode_near_result(const near_result_t *result,
```

Reply:
Updated